modules/programs/sunpaper.nix

```diff
@@ -0,0 +1,70 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.sunpaper;
+  newDerivation = (pkgs.sunpaper.overrideAttrs (previousAttrs: {
+    postPatch = concatStrings [
+      previousAttrs.postPatch
+      ''
+        substituteInPlace sunpaper.sh \
+          --replace "38.9072N" ${cfg.latitude} \
+          --replace "77.0369W" ${cfg.longitude} \
+          --replace "Corporate-Synergy" "${cfg.wallpaper_collection}"
+      ''
+    ];
+  }));
```

Comment:
Seems better to create a [configuration file](https://github.com/hexive/sunpaper/blob/8d518dfddb5e80215ef3b884ff009df1d4bb74c2/sunpaper.sh#L199-L203) instead of trying to patch the script?